onelogin_rules/onelogin_active_login_activity.py

```diff
@@ -0,0 +1,36 @@
+import time
+from panther_oss_helpers import get_string_set, put_string_set, set_key_expiration, add_to_string_set  # pylint: disable=import-error,line-too-long
+
+THRESH = 2
+THRESH_TTL = 43200  # 1/2 day
+
+
+def rule(event):
+    # Pre-filter: event_type_id = 5 is login events.
+    if event.get('event_type_id') != 5 or not event.get(
+            'ipaddr') or not event.get('user_id'):
+        return False
+    # This track multiple successful logins for different accounts from the same ip address
+    # First, keep a list of unique usernames that have logged in from this ip address
```

Comment:
I like this rule! The only consideration is that when everyone is in the same corporate office, they'll likely be sharing a small subset of public IPs. It may be worthwhile to put in a whitelist here such as `CORPORATE_PUBLIC_IP_SPACE` or something similar, and IP addresses in that range get ignored. This whitelist could also be added as a global if you feel it could be generically re-useable.

Reply:
That makes sense. I added a new helper function in the panther_oss_helpers that is a generic "is this ip in this list of subnets"? called `is_ip_in_network` and created a list called `SHARED_IP_SPACE` - since this could also contain proxy servers (?).  I'm not loving the name, though...

Reply:
Having a function and a list makes sense. I think `SHARED_IP_SPACE` gets the idea across, as long as you have a comment nearby explaining what it's for it should be fine.